protocol/communities_messenger_token_permissions_test.go

```diff
@@ -2054,6 +2054,59 @@ func (s *MessengerCommunitiesTokenPermissionsSuite) TestResendEncryptionKeyOnBac
 	s.Require().Len(response.Messages(), 1)
 }
 
+func (s *MessengerCommunitiesTokenPermissionsSuite) TestResendSharedAddressesOnBackupRestore() {
+	community, _ := s.createCommunity()
+
+	// bob joins the community
+	s.advertiseCommunityTo(community, s.bob)
+	s.joinCommunity(community, s.bob, bobPassword, []string{})
+
+	currentBobSharedAddresses, err := s.bob.GetRevealedAccounts(community.ID(), s.bob.IdentityPublicKeyString())
+
+	// Simulate backup creation and handling backup message
+	// As a result, bob sends request to resend encryption keys to the owner
+	clock, _ := s.bob.getLastClockWithRelatedChat()
+
+	community, err = s.owner.communitiesManager.GetByID(community.ID())
+	s.Require().NoError(err)
+
+	backupMessage, err := s.bob.backupCommunity(community, clock)
+	s.Require().NoError(err)
+
+	err = s.bob.HandleBackup(s.bob.buildMessageState(), backupMessage, nil)
+	s.Require().NoError(err)
+
+	// FIXME this is necessary to have the owner receive the request and there is no response
+	_, err = WaitOnMessengerResponse(
+		s.owner,
+		func(r *MessengerResponse) bool {
+			if len(r.requestsToJoinCommunity) > 0 && r.requestsToJoinCommunity[community.IDString()] != nil {
+				return true
+			}
+			return false
+		},
+		"request to join not received",
+	)
```

Comment:
@mprakhov can you give me a hint how to solve this? `handleCommunitySharedAddressesRequest` doesn't have any response, so I don't know what to wait for. Can we hook to the message ending and wait for the response to have been sent?

Reply:
I just tried using Subscriptions, but it doesn't work for me, because I need to trigger `RetrieveAll` for the Owner.

I ended up using `BackupHandled` on the Response to say when the owner handled the response.

It's not super clean, but it works. If you prefer, I can introduce a new property on the MessengerResponse, but it would only be for tests.

Reply:
Do you think I should redo my code/test to use that or using the MessengerResponse is good enough?

Reply:
Opened here: https://github.com/status-im/status-go/issues/5326